interview-questions.md

```diff
@@ -8,36 +8,90 @@ Then, research the question to expand on your answer. Even if you feel you have
+Secondly, generate a migration from your terminal with rails db:migrate add_foreign_key_column. This will generate a migration file that you can use the add_column line to generate your foreign key column. You will want to add this column to your :students table, and should name it :cohort_id to properly set up your rails configuration. 
+After this, you need to run a rails db:migrate command to complete your migration, then you can add in the appropriate keys and you are good to go!
 
 Researched answer:
+When using the has_many/belongs_to assosciation, the model that belongs_to needs to have a column to contain the foreign key. The foreign key needs to be in the format of model_id with model being the singular of the model. 
+
+Each class needs to have the association added into their class definitions (inside app/models) added into them. The correct format for the class that has_many is has_many :models where models is the model that it has many of (reminder that this needs to be the plural of the word). The class that belongs_to needs to have the line belongs_to :model where model is the singular of the model it belongs_to
+
+If creating an assosciation after the initial model generation, a migration can be generated, then the add_column method can be used to add this column
+
 
 2. Which RESTful routes must always be passed params? Why?
 
 Your answer:
+Show, edit, update, and destroy all need to have params passed to them. Specifically they need to have the :id parameter in order to access the proper item. Technically speaking, it is possible to have your database set up to not need parameters in order to create a new instance. 
 
 Researched answer:
+The Show, edit, update, and destroy RESTful routes need to have params passed to them as the route points to a specfic instance. Without passing the parameter, there is no way to confirm which instance is having the route called on it. 
 
 3. Name three rails generator commands. What is created by each?
 
 Your answer:
+generate model, which creates a table
+generate migration, which creates a migration file you can use to update your database
+generate resource, which creates a model and RESTful routes for basic CRUD actions
 
 Researched answer:
+Rails generators are used for streamlining workflow by generating multiple files structures for commonly used routing, views, and controllers. While generators are incredibly helpful, it is also important to understand what they are generating, as well as knowing when to use and not use them. 
+The generate controller command is a great example, it generates a controller with every method that you add to the command. The issue is that it ALSO generates a view for each method. This is often unneeded, so if you are going to use this generator, be sure to only use it to generate methods that require a view, or be sure to go in and remove the unneeded views. 
```

Comment:
Love the consideration expressed here with being mindful of the files we are creating with a command. 👏🏽